src/processes/follow-topic-process.js

```diff
@@ -122,7 +164,11 @@ const followTopicProcess = async (job, done) => {
         return;
       }
       // check if user is anonymous
-      const senderUser = await User.findByPk(communityMessageFormat.user_id);
+      const [senderUser, receiverUser] = await Promise.all([
+        User.findByPk(communityMessageFormat.user_id),
+        User.findByPk(data.user_id)
+      ]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The Promise.all() function is used here to fetch senderUser and receiverUser concurrently, which can improve performance. However, if one promise fails, Promise.all() will reject with the reason of the first promise that rejects. This means if fetching either senderUser or receiverUser fails, we won't know about the status of the other. If you want to ensure that both promises are fulfilled or rejected, consider using Promise.allSettled() instead.



<!-- This is an auto-generated comment by OSS CodeRabbit -->